### PR TITLE
Add Compatiblity Datasource and Resource

### PR DESCRIPTION
**Pull request by doriac11**
# Description
- Added Resource and Datasource for Compatibility Management.

# Checklist:

- [x] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [x] I have verified that new and existing unit tests pass locally with my changes
- [x] I have not allowed coverage numbers to degenerate
- [x] I have maintained at least 80% code coverage
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] Backward compatibility is not broken

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Please also list any relevant details for your test configuration

- [x] Acceptance Tests
## Output
image

## Coverage Report
image

image

### CODE REVIEW

File: powerflex/helper/compatibility_management_helper.go [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+	data, err := os.ReadFile(plan.RepositoryPath.ValueString())
+	if err != nil {
+		return nil, fmt.Errorf("Could not read repository file, make sure path to gpg file is correct: " + err.Error())
+	}
+
+	return client.SetCompatibilityManagement(&scaleiotypes.CompatibilityManagementPost{
+		CompatibilityDataBytes: data,
+		RepositoryPath:         plan.RepositoryPath.ValueString(),
+		// Source is always hardcoded to local, that is the only supported value in PowerFlex
+		Source: "local",
```
Comment (AnikaAgiwal2711) [inline powerflex/helper/compatibility_management_helper.go:50]: Do we need to hardcode this?? Do we need to handle CIFS share cases here

Comment (doriac11) [reply powerflex/helper/compatibility_management_helper.go]: Hey @AnikaAgiwal2711, the only supported source here is local, we confirmed this with the Powerflex team.

**Review by nikitajoshi1 — DISMISSED** [review]

File: docs/data-sources/compatibility_management.md [hunk 2]
```diff
@@ -0,0 +1,75 @@
+---
+# Copyright (c) 2023-2024 Dell Inc., or its subsidiaries. All Rights Reserved.
```
Comment (shenda1) [inline docs/data-sources/compatibility_management.md:2]: 2024

File: powerflex/provider/compatibility_management_resource_schema.go [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+
+// CompatibilityManagementResourceSchema defines the schema for device datasource
+var CompatibilityManagementResourceSchema schema.Schema = schema.Schema{
+	Description:         "This resource is used to set the g compatibility management from the PowerFlex array.",
+	MarkdownDescription: "This resource is used to set the g compatibility management from the PowerFlex array.",
+	Attributes: map[string]schema.Attribute{
+		"id": schema.StringAttribute{
+			Description:         "Unique identifier of the Compatibility Management Instance.",
+			MarkdownDescription: "Unique identifier of the Compatibility Management Instance.",
+			Computed:            true,
```
Comment (rounak-adhikary) [inline powerflex/provider/compatibility_management_resource_schema.go:32]: Please check if any of these `Optional+Computed` attributes should be only `Computed`. Looks to me like the user only inputs `repository_path` and all else are purely computed.

File: powerflex/provider/compatiblity_management_datasource_schema.go [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+
+// CompatibilityManagementDataSourceSchema defines the schema for device datasource
+var CompatibilityManagementDataSourceSchema schema.Schema = schema.Schema{
+	Description:         "This datasource is used to query the existing compatibility management from the PowerFlex array. The information fetched from this datasource can be used for getting the details / for further processing in resource block.",
+	MarkdownDescription: "This datasource is used to query the existing compatibility management from the PowerFlex array. The information fetched from this datasource can be used for getting the details / for further processing in resource block.",
+	Attributes: map[string]schema.Attribute{
+		"id": schema.StringAttribute{
+			Description:         "Unique identifier of the Compatibility Management Instance.",
+			MarkdownDescription: "Unique identifier of the Compatibility Management Instance.",
+			Optional:            true,
```
Comment (rounak-adhikary) [inline powerflex/provider/compatiblity_management_datasource_schema.go:32]: Isnt this computed?

File: powerflex/provider/compatibility_management_resource_schema.go [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+
+package provider
+
+import (
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+)
+
+// CompatibilityManagementResourceSchema defines the schema for device datasource
+var CompatibilityManagementResourceSchema schema.Schema = schema.Schema{
+	Description:         "This resource is used to set the g compatibility management from the PowerFlex array.",
```
[the remark this answers is not in the document]
Comment (doriac11) [reply powerflex/provider/compatibility_management_resource_schema.go:26]: Hey Krunal, yes that was a typo, will make this fix.

File: docs/data-sources/compatibility_management.md (hunk 2 above)
Comment (doriac11) [reply docs/data-sources/compatibility_management.md:2]: @shenda1  will make the update to 2024

File: powerflex/provider/compatibility_management_resource_schema.go (hunk 3 above)
Comment (doriac11) [reply powerflex/provider/compatibility_management_resource_schema.go:32]: Made the updates.

**Review by rounak-adhikary — DISMISSED** [review]

**Comment by doriac11** [comment]
run e2e test

**Comment by doriac11** [comment]
run e2e test TestAccCompatibilityManagementResource

**Review by nikitajoshi1 — DISMISSED** [review]

**Comment by doriac11** [comment]
run e2e test TestAccCompatibilityManagementResource

**Review by nikitajoshi1 — APPROVED** [review]

**Review by shenda1 — APPROVED** [review]

**Pull request merged**